pkg/validation/options.go

```diff
@@ -60,117 +68,123 @@ func Validate(o *options.Options) error {
 			"\n      use email-domain=* to authorize all email addresses")
 	}
 
-	if o.Providers[0].OIDCConfig.IssuerURL != "" {
+	o = o.InitProviders()
```

Comment:
While we are touching this validation code, is it possible to extract the validation for providers into a separate file so that it's all self contained?

Are there options outside of the provider that are required to enable each provider validation?

You can see the pattern we are aiming for if you look at the other files in this package